### ISSUE TITLE
Cityscapes task and performance

### ISSUE THREAD

**Issue by reactivetype** [ultralytics/yolov5#1158]
Has anyone tried this repo on Cityscapes detection? I am curious what the AP performance are for the models.

Perhaps, it would be nice to support cityscapes in addition to VOC and COCO

**Comment by glenn-jocher** [comment]
@reactivetype Cityscapes and KITTI are two datasets we receive frequent requests for. Cityscapes appears to be entirely for segmentation tasks however, whereas KITTI, VOC and COCO include detection tasks that YOLOv5 is natively set up for.

I don't have experience myself with segmentation, but if you'd like to submit a PR for this we'd be happy to work with you to get it merged. I imagine the additions required would be extensive: dataloader, loss functions, inference, mAP computation, plotting tools.

**Comment by glenn-jocher** [comment]
@reactivetype I forgot to mention model architecture changes. You'd want to replace either the entire head or at least the Detect() module with your own custom designed Segment() module, ideally reusing whatever SOTA segmentation heads exist at the moment.

https://github.com/ultralytics/yolov5/blob/3b57cb56412976847b91ef184ca2f2a99491d8ab/models/yolov5s.yaml#L27-L48

**Comment by reactivetype** [comment]
@glenn-jocher  Thanks for the reply. It was not clear in my request that I meant the task was object detection and not segmentation. You are right that cityscapes is also a dataset for semantic segmentation. However, each image also comes with bounding box annotations. 
If the task is object detection, are the only changes required the model configuration (number of classes, anchors) and data preparation? For example, a get_cityscapes.sh script to prepare the data in YOLO format.
How about the AP evaluation? Does it require some changes?

**Comment by glenn-jocher** [comment]
@reactivetype ah got it. In that case all we need is a `cityscapes.yaml` file for the dataset, and ideally a short bash script we could use to download and convert to YOLO, i.e. `get_cityscapes.sh`. Both of these would go in the yolov5/data folder.

For training/testing/inference/export no changes are required, i.e. to train cityscapes once the above steps are done:
```bash
python train.py --weights yolov5m.pt --data cityscapes.yaml 
```

**Comment by glenn-jocher** [comment]
@reactivetype for label conversion to YOLO format see https://docs.ultralytics.com/yolov5/tutorials/train_custom_data#2-create-labels

**Comment by glenn-jocher** [comment]
@reactivetype any progress? If you have any other questions let me know.
